### PR TITLE
DC'd players no longer bamboozle first responders

### PR DESCRIPTION
**Pull request by EntranceJew**
- relocated MakeSpawnPointSafe since there are cheaper checks ahead of
  where it gets used
- also added a hook to let people decline defib requests for their own reasons

### CODE REVIEW

**Comment by EntranceJew** [comment]
i should mention that originally without this IsValid check, it hard-errors server-side

File: lua/terrortown/lang/en/defibrillator.lua [hunk 1]
```diff
@@ -15,6 +15,7 @@ L["defi_error_no_valid_ply"] = "You can't revive this player since they are no l
 L["defi_error_already_reviving"] = "You can't revive this player since they are already reviving."
 L["defi_error_failed"] = "Revival attempt failed. Please try again."
 L["defi_error_player_alive"] = "You can't revive this player since they are already alive."
+L["defi_error_player_disconnected"] = "You can't revive this player since they have disconnected."
```
Comment (saibotk) [inline lua/terrortown/lang/en/defibrillator.lua:18]: Can you copy this to the other language files and then comment them out in those?

File: gamemodes/terrortown/entities/weapons/weapon_ttt_defibrillator.lua [hunk 2]
```diff
@@ -277,14 +282,32 @@ if SERVER then
...
+		end
+
+		local corpsePlayer = CORPSE.GetPlayer(ent)
+		if not IsValid(corpsePlayer) then
+			-- In case people want to do something about this for themselves.
+			local defibDisconnectResult = hook.Run("TTT2AttemptDefibDisconnectedPlayer", owner, ent, self)
+			if defibDisconnectResult ~= nil then
+				self:Error()
+				return
+			end
```
Comment (saibotk) [inline gamemodes/terrortown/entities/weapons/weapon_ttt_defibrillator.lua:298]: I feel like this is very specific and would rather see this as a general hook in the error method to let people react to all error events.

**Review by saibotk — CHANGES_REQUESTED** [review]

File: lua/terrortown/lang/en/defibrillator.lua (hunk 1 above)
Comment (EntranceJew) [reply lua/terrortown/lang/en/defibrillator.lua:18]: also handled in follow-up

**Review by saibotk — APPROVED** [review]
Looks good 👍 thank you 🙏

**Pull request merged**